mujoco_warp/_src/solver.py

```diff
@@ -101,18 +103,39 @@ def _efc_kernel(d: types.Data):
     Jaref = d.efc.Jaref[efcid]
     efc_D = d.efc.D[efcid]
 
-    # TODO(team): active and conditionally active constraints
-    active = int(Jaref < 0.0)
-    d.efc.active[efcid] = active
+    active = (Jaref < 0.0) or (efcid < d.ne[0] + d.nf[0])
```

Comment:
This looks like the JAX-y no-if version of this code, I'm wondering if Warp can let us do:

```py
ne = d.ne[0]
nf = d.nf[0]
if efcid < ne:
  # equality constraints
  pass
elif efcid < ne + nf:
  # friction
  ...
else:
  # contact
  active = Jaref < 0.0
  ...
```

Benchmark it though to make sure it doesn't slow down the step!

Reply:
updated the code to use if statements